application/console/migrations/m181025_155801_create_method_table.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use yii\db\Migration;
+
+/**
+ * Handles the creation of table `method`.
+ */
+class m181025_155801_create_method_table extends Migration
+{
+    /**
+     * @inheritdoc
+     */
+    public function up()
+    {
+        $this->createTable(
+            '{{method}}',
+            [
+                'id' => 'pk',
+                'uid' => 'char(32) not null',
+                'user_id' => 'int(11) not null',
+                'value' => 'varchar(255) not null',
+                'verified' => 'boolean not null default false',
+                'verification_code' => 'varchar(64) null',
+                'verification_attempts' => 'smallint null',
+                'verification_expires' => 'datetime null',
+                'created' => 'datetime not null',
+            ],
+            "ENGINE=InnoDB DEFAULT CHARSET=utf8"
+        );
+        $this->addForeignKey('fk_method_user_id','{{method}}','user_id','{{user}}','id','CASCADE','NO ACTION');
```

Comment:
@fillup, what did we decide to do regarding CASCADE vs. NO ACTION for on-delete and on-update? I think we've been using the Yii model's `beforeDelete()` to individually delete child records (since that lets us enforce any `beforeDelete()` business rules for the child records), but I'm not certain.

Reply:
I added cascading delete code to `beforeDelete()` in the User model, and changed the `$delete` parameter of `addForeignKey` to `NO ACTION`.